### PR TITLE
rusk-wallet-2622 | Set default gas price to the average mempool gas price

### PR DESCRIPTION
**Pull request by welf**
Closes [#2622 ](https://github.com/dusk-network/rusk/issues/2622)

- [x] Add `MempoolGasPrices` struct 
- [x] Provide an access to the HTTP client to the `Wallet` struct and its `call` method
- [x] Add `get_mempool_gas_prices` method to the `Wallet`
- [x] Refactor `prompt::request_gas_price` function to use the average gas price from the mempool as the default gas price value
- [x] Update calls of the `prompt::request_gas_price` function

### CODE REVIEW

File: rusk-wallet/src/gas.rs [hunk 1]
```diff
@@ -89,3 +91,16 @@ impl Default for Gas {
         Self::new(DEFAULT_LIMIT_TRANSFER)
     }
 }
+
+/// Dynamic gas prices information from the mempool
+#[derive(Debug, Deserialize)]
+pub struct MempoolGasPrices {
+    /// Average gas price in the mempool in [Lux]
+    pub average: Lux,
+    /// Maximal gas price in the mempool in [Lux]
```
Comment (Neotamandua) [inline rusk-wallet/src/gas.rs:100]:
```suggestion
    /// Maximum gas price in the mempool in [Lux]
```

**Review by Neotamandua — CHANGES_REQUESTED** [review]
Generally looks good so far! I suggested a small comment change. 

In addition, can you squash your commits into "rusk-wallet: ...". We have a convention of mentioning which member of the workspace is being changed by the commit. You can also look at the commit history for reference.

File: rusk-wallet/src/gas.rs (hunk 1 above)
Comment (welf) [reply rusk-wallet/src/gas.rs]: Done

**Review by Neotamandua — APPROVED** [review]
LGTM

**Pull request merged**